airbyte-cdk/python/unit_tests/sources/declarative/requesters/request_options/test_interpolated_request_options_provider.py

```diff
@@ -59,6 +59,9 @@ def test_interpolated_request_params(test_name, input_request_params, expected_r
         ("test_number_falsy_value", {"number_falsy": "{{ 0.0 }}"}, {"number_falsy": 0.0}),
         ("test_string_falsy_value", {"string_falsy": "{{ '' }}"}, {}),
         ("test_none_value", {"none_value": "{{ None }}"}, {}),
+        ("test_string", """{"nested": { "key": "{{ config['option'] }}" }}""", {"nested": {"key": "OPTION"}}),
```

Comment:
nit: can you add a test case with a list and another one with a number to cover the branches in https://github.com/airbytehq/airbyte/pull/26474/files#diff-e94b83408916b273194cf715b73cb227bee59ecc69bb2f8148a3cc62a7162952R42-R43 and https://github.com/airbytehq/airbyte/pull/26474/files#diff-e94b83408916b273194cf715b73cb227bee59ecc69bb2f8148a3cc62a7162952R44-R45

Reply:
doesn't the union `NestedMappingEntry` include numbers and lists? https://github.com/airbytehq/airbyte/pull/26474/files#diff-e94b83408916b273194cf715b73cb227bee59ecc69bb2f8148a3cc62a7162952R12

Reply:
The entries yes, but the top level “mapping” parameter that needs to be passed in doesn’t